test/base/src/androidTest/kotlin/io/realm/test/shared/dynamic/DynamicMutableRealmTests.kt

```diff
@@ -79,60 +85,192 @@ class DynamicMutableRealmTests {
 
     // TODO Add variants for each type
     @Test
-    fun createPrimaryKey() {
+    fun copyToRealm_withPrimaryKey() {
         val dynamicMutableObject =
-            dynamicMutableRealm.createObject("PrimaryKeyString", "PRIMARY_KEY")
+            dynamicMutableRealm.copyToRealm(DynamicMutableRealmObject.create("PrimaryKeyString", "primaryKey" to "PRIMARY_KEY"))
         assertTrue { dynamicMutableObject.isValid() }
         assertEquals("PRIMARY_KEY", dynamicMutableObject.getValue("primaryKey"))
     }
 
     // TODO Add variants for each type
     @Test
-    fun createPrimaryKey_nullablePrimaryKey() {
-        val dynamicMutableObject =
-            dynamicMutableRealm.createObject("PrimaryKeyStringNullable", null)
+    fun copyToRealm_withPrimaryKey_null() {
+        val dynamicMutableObject = dynamicMutableRealm.copyToRealm(DynamicMutableRealmObject.create("PrimaryKeyStringNullable", "primaryKey" to null))
         assertTrue { dynamicMutableObject.isValid() }
         assertNull(dynamicMutableObject.getNullableValue<String>("primaryKey"))
     }
 
     @Test
-    fun create_throwsOnUnknownClass() {
+    fun copyToRealm_tree() {
+        val child = DynamicMutableRealmObject.create("Sample", "stringField" to "CHILD")
+        val obj = DynamicMutableRealmObject.create(
+            "Sample",
+            "stringField" to "PARENT",
+            "stringListField" to realmListOf("1", "2", "3"),
+            "nullableObject" to child,
+            "objectListField" to realmListOf(child, child, child),
+        )
+        dynamicMutableRealm.copyToRealm(obj)
+
+        dynamicMutableRealm.query("Sample", "stringField = 'PARENT'").find().single().run {
+            assertEquals(listOf("1", "2", "3"), getValueList("stringListField"))
+            assertEquals("CHILD", getObject("nullableObject")!!.getValue("stringField"))
+            getObjectList("objectListField").run {
+                assertEquals(3, size)
+                forEach { assertEquals("CHILD", it.getValue("stringField")) }
+            }
+        }
+        val managedChildren = dynamicMutableRealm.query("Sample", "stringField = 'CHILD'").find()
+        assertEquals(1, managedChildren.size)
+    }
+
+    @Test
+    fun copyToRealm_withManagedDynamicObject() {
+        val child = dynamicMutableRealm.copyToRealm(
+            DynamicMutableRealmObject.create(
+                "Sample",
+                "stringField" to "CHILD",
+            )
+        )
+        dynamicMutableRealm.copyToRealm(
+            DynamicMutableRealmObject.create(
+                "Sample",
+                "stringField" to "PARENT",
+                "nullableObject" to child
+            )
+        )
+        dynamicMutableRealm.query("Sample", "stringField = 'PARENT'").find().single().run {
+            getObject("nullableObject")!!.run {
+                assertEquals("CHILD", getValue("stringField"))
+            }
+        }
+    }
+
+    @Test
+    fun copyToRealm_withUnmanagedTypedObject() {
```

Comment:
You mean dynamic object?

Reply:
No. We can also import typed objects, so added a test for that 🤷 ... is it too confusing that we allow that? 🤔

Reply:
If we allow this is there a test that makes sure they're using the same version?

Reply:
So, you would like to enforce that `schemaVersion` is the same for both? I don't think we have a way to enforce that, since there is no schema version tied to the unmanaged object. The schema of that is exactly whats in the model definition. I guess we could just iterate the class metadata for it but it sounds like a bit effort. If people do it I would assume that they know what they do ... maybe better to block this completely ... but it is just awfully convenient to be able to just create objects from the typed one in a dynamic realm!!??

Reply:
IMO the current behavior seems fine. Dynamic objects are by their nature a little more brittle and depend on people knowing what they are doing